gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/aggregation/functions/AggregateUtil.java

```diff
@@ -30,37 +28,21 @@
 class AggregateUtil {
 
   /**
-   * Increments the aggregate map by the increment of the aggregate functions on the vertex
+   * Increments the aggregate map by the increment of the aggregate functions on the element
    *
    * @param aggregate aggregate map to be incremented
-   * @param vertex vertex to increment with
+   * @param element element to increment with
    * @param aggregateFunctions aggregate functions
+   * @param <T> element type
    * @return incremented aggregate map
    */
-  static Map<String, PropertyValue> vertexIncrement(Map<String, PropertyValue> aggregate,
-    Vertex vertex, Set<VertexAggregateFunction> aggregateFunctions) {
-    for (VertexAggregateFunction aggFunc : aggregateFunctions) {
-      PropertyValue increment = aggFunc.getVertexIncrement(vertex);
-      if (increment != null) {
-        aggregate.compute(aggFunc.getAggregatePropertyKey(), (key, agg) -> agg == null ?
-          increment.copy() : aggFunc.aggregate(agg, increment));
-      }
-    }
-    return aggregate;
-  }
-
-  /**
-   * Increments the aggregate map by the increment of the aggregate functions on the edge
-   *
-   * @param aggregate aggregate map to be incremented
-   * @param edge edge to increment with
-   * @param aggregateFunctions aggregate functions
-   * @return incremented aggregate map
-   */
-  static Map<String, PropertyValue> edgeIncrement(Map<String, PropertyValue> aggregate, Edge edge,
-    Set<EdgeAggregateFunction> aggregateFunctions) {
-    for (EdgeAggregateFunction aggFunc : aggregateFunctions) {
-      PropertyValue increment = aggFunc.getEdgeIncrement(edge);
+  @SuppressWarnings("unchecked")
+  static <T extends Element> Map<String, PropertyValue> increment(
+    Map<String, PropertyValue> aggregate,
+    T element,
+    Set<AggregateFunction> aggregateFunctions) {
```

Comment:
I don't really like this typeless use here, not sure how to handle this though...